pypeit/core/telluric.py

```diff
@@ -617,132 +637,136 @@ def general_spec_reader(specfile, ret_flam=False):
-        wave (`numpy.ndarray`_): wavelength in units of Angstrom
-        flux (`numpy.ndarray`_): flux array
-        ivar (`numpy.ndarray`_): inverse variance array.
-        mask (`numpy.ndarray`_): good pixel mask for your spectrum.
-        spectrograph (str): spectrograph name
-        telluric (`numpy.ndarray`_): telluric model
-        obj_model (`numpy.ndarray`_): object model used for telluric fitting
-        header (`astropy.io.fits.Header`): Primary header
-        ex_value (str): extraction mode, OPT or BOX
-        overwrite (bool):
+        outfile (:obj:`str`):
+            File to output telluric corrected spectrum to.
+        wave (`numpy.ndarray`_):
+            wavelength in units of Angstrom
+        flux (`numpy.ndarray`_):
+            flux array
+        ivar (`numpy.ndarray`_):
+            inverse variance array.
+        mask (`numpy.ndarray`_):
+            good pixel mask for your spectrum.
+        spectrograph (:obj:`str`, optional):
+            spectrograph name
+        telluric (`numpy.ndarray`_, optional):
+            telluric model
+        obj_model (`numpy.ndarray`_, optional):
+            object model used for telluric fitting
+        header (`astropy.io.fits.Header`_, optional):
+            Primary header
+        ex_value (:obj:`str`, optional):
+            extraction mode, OPT or BOX
+        overwrite (:obj:`bool`, optional):
            Overwrite existing file?
     """
-
     wave_mask = wave > 1.0
-    # Generate the DataContainer
-    onespec = coadd1d.OneSpec(wave[wave_mask],
-                      flux[wave_mask],
-                      PYP_SPEC=spectrograph,
-                      ivar=ivar[wave_mask],
-                      mask=mask[wave_mask].astype(int),
-                      ext_mode=ex_value,
-                      fluxed=True)
-    onespec.head0 = header
-
-    # Add on others
-    if telluric is not None:
-        onespec.telluric = telluric[wave_mask]
-    if obj_model is not None:
-        onespec.obj_model = obj_model[wave_mask]
-    # Write
-    onespec.to_file(outfile, overwrite=overwrite)
-
-############################
-#  Object model functions  #
-############################
-
+    spec = onespec.OneSpec(wave[wave_mask], flux[wave_mask], PYP_SPEC=spectrograph,
+                           ivar=ivar[wave_mask], mask=mask[wave_mask].astype(int),
+                           telluric=None if telluric is None else telluric[wave_mask],
+                           obj_model=None if obj_model is None else obj_model[wave_mask],
+                           ext_mode=ex_value, fluxed=True)
+    # TODO: We need a better way of assigning the header...
+    spec.head0 = header
+    spec.to_file(outfile, overwrite=overwrite)
 
 
 ##############
 # Sensfunc Model #
 ##############
-def init_sensfunc_model(obj_params, iord, wave, counts_per_ang, ivar, gpm, tellmodel):
+def init_sensfunc_model(obj_params, iord, wave, counts_per_ang, ivar, mask, tellmodel):
```

Comment:
Is there a reason why you've renamed gpm mask? I thought we were moving towards a convention where we always used gpm instead of mask, which is what I was aiming for here. Also, you changed mask, but did not change the name in the docs.

Reply:
Thanks for catching this!  This must have been a bad merge on my part.  I've fixed this.